docs/configs.md

```diff
@@ -36,6 +36,7 @@ Name | Description | Default Value
 <a name="memory.gpu.reserve"></a>spark.rapids.memory.gpu.reserve|The amount of GPU memory that should remain unallocated by RMM and left for system use such as memory needed for kernels, kernel launches or JIT compilation.|1073741824
 <a name="memory.host.spillStorageSize"></a>spark.rapids.memory.host.spillStorageSize|Amount of off-heap host memory to use for buffering spilled GPU data before spilling to local disk|1073741824
 <a name="memory.pinnedPool.size"></a>spark.rapids.memory.pinnedPool.size|The size of the pinned memory pool in bytes unless otherwise specified. Use 0 to disable the pool.|0
+<a name="shuffle.fetchTimeout"></a>spark.rapids.shuffle.fetchTimeout|Timeout that the RapidsShuffleIterator will wait for a resolved batch. It defaults to 120s since this is the value that Spark uses for `spark.network.timeout`.|120
```

Comment:
Why not use `spark.network.timeout` for the timeout value?  Do we really need a separate config for the same concept?

Reply:
`spark.network.timeout` is a private config within spark, from the code. I can try using that for our timeout, but this is why I didn't just get `spark.network.timeout`. If that's OK for us, then I can change it to use it.

Reply:
It's not really private, as it's clearly documented here for users to use: https://spark.apache.org/docs/latest/configuration.html.  Maybe the variable that holds the config key within Spark is private, but we don't need to reuse that key variable, we can just place this literal string in our code because it's part of the formal Spark API.  They can't remove support for this config without breaking users that rely on being able to change this setting.

Reply:
We do have a couple of other places that could be `connectionTimeout`. It seems like you want that setting to be pretty small, IMHO, with relation to a timeout for fetches. The fetch timeout could be larger in reality, if you are accounting for busy servers, so my vote would be to use `spark.network.timeout` or our own setting. Given discussion here, I'll move to the `spark.network.timeout` for now. If we think we should use the `spark.shuffle.io.connectionTimeout` that can happen in this PR, or a subsequent one. Just let me know

Reply:
So I changed to use `spark.network.timeout` and while testing I am running into this:

```
java.lang.IllegalArgumentException: requirement failed: The value of spark.network.timeout=5000ms must be no less than the value of spark.executor.heartbeatInterval=10000ms.
```

One potential issue with using this setting, is that the user needs to watch out for the `heartbeatInterval` timeout in the process. 120 seconds is ridiculously high for what we need, so maybe this is OK, as there's a lot of room between 10 seconds (`heartbeatInterval` default) and 120 seconds. Just mentioning it here anyone has concerns.

Reply:
My main point here is that if we can find a Spark config that mirrors what we're trying to measure then we should use it if possible.  Maybe `spark.network.timeout` isn't always appropriate, and if there isn't another alternative then maybe it justifies creating a config specific to the plugin.

I just don't want to add configs that are redundant with the standard Spark configs users are already familiar with if we can sensibly reuse those.

Reply:
> I just don't want to add configs that are redundant with the standard Spark configs users are already familiar with if we can sensibly reuse those.

I agree. This seems like a good place to start (and better than where we are now), and if we run into issues with it down the line we can pick something else, or add our own config citing why it doesn't work as planned.

Reply:
Updated the branch.